astronomer/providers/amazon/aws/hooks/batch_client.py

```diff
@@ -0,0 +1,256 @@
+
+    async def poll_for_job_running(  # type: ignore[override]
+        self, job_id: str, delay: Union[int, float, None] = None
+    ) -> None:
+        """
+        Poll for job running. The status that indicates a job is running or
+        already complete are: 'RUNNING'|'SUCCEEDED'|'FAILED'.
+
+        So the status options that this will wait for are the transitions from:
+        'SUBMITTED'>'PENDING'>'RUNNABLE'>'STARTING'>'RUNNING'|'SUCCEEDED'|'FAILED'
+
+        The completed status options are included for cases where the status
+        changes too quickly for polling to detect a RUNNING status that moves
+        quickly from STARTING to RUNNING to completed (often a failure).
+
+        :param job_id: a Batch job ID
+
+        :param delay: a delay before polling for job status
+
+        :raises: AirflowException
+        """
+        await self.delay(delay)
+        running_status = [self.RUNNING_STATE, self.SUCCESS_STATE, self.FAILURE_STATE]
+        await self.poll_job_status(job_id, running_status)
+
+    async def get_job_description(self, job_id: str) -> Dict[str, str]:  # type: ignore[override]
+        """
+        Get job description (using status_retries).
+
+        :param job_id: a Batch job ID
+
+        :return: an API response for describe jobs
+
+        :raises: AirflowException
+        """
+        retries = 0
+        async with await self.get_client_async() as client:
+            while True:
+                try:
+                    response = await client.describe_jobs(jobs=[job_id])
+                    return self.parse_job_description(job_id, response)
+
+                except botocore.exceptions.ClientError as err:
+                    error = err.response.get("Error", {})
+                    if error.get("Code") == "TooManyRequestsException":
+                        pass  # allow it to retry, if possible
+                    else:
+                        raise AirflowException(f"AWS Batch job ({job_id}) description error: {err}")
+
+                retries += 1
+                if retries >= self.status_retries:
+                    raise AirflowException(
+                        f"AWS Batch job ({job_id}) description error: exceeded status_retries "
+                        f"({self.status_retries})"
+                    )
+
+                pause = self.exponential_delay(retries)
+                self.log.info(
+                    "AWS Batch job (%s) description retry (%d of %d) in the next %.2f seconds",
+                    job_id,
+                    retries,
+                    self.status_retries,
+                    pause,
+                )
+                await self.delay(pause)
+
+    async def poll_job_status(self, job_id: str, match_status: List[str]) -> bool:  # type: ignore[override]
+        """
+        Poll for job status using an exponential back-off strategy (with max_retries).
+
+        :param job_id: a Batch job ID
+
```

Comment:
you can remove this extra blank line

Reply:
done in [f8c026d](https://github.com/astronomer/astronomer-providers/pull/404/commits/f8c026d9b7dccc417b17fd158282d0b265290ecf)